astropy/io/fits/scripts/fitsheader.py

```diff
@@ -275,9 +292,76 @@ def print_headers_as_table(args):
     resulting_table.write(sys.stdout, format=args.table)
 
 
+def print_headers_as_comparison(args):
+    """Prints FITS header(s) with keywords as columns.
+    
+    This follows the dfits+fitsort format.
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        Arguments passed from the command-line as defined below.
+    """
+    tables = []
+    # Create a Table object for each file
+    for filename in args.filename:  # Support wildcards
+        try:
+            formatter = TableHeaderFormatter(filename)
+            tbl = formatter.parse(args.extensions,
+                                  args.keywords,
+                                  args.compressed)
+            if tbl:
+                tables.append(tbl)
+        except OSError as e:
+            log.error(str(e))  # file not found or unreadable
+    # Concatenate the tables
+    if len(tables) == 0:
+        return False
+    elif len(tables) == 1:
+        resulting_table = tables[0]
+    else:
+        from .... import table
+        resulting_table = table.vstack(tables)
+    # If we obtained more than one hdu, merge hdu and keywords columns
+    if len(np.unique(resulting_table['hdu']))>1:
+        for tab in tables:
+            new_column = table.Column(['{}:{}'.format(row['hdu'],row['keyword']) for row in tab])
+            tab.add_column(new_column, name='hdu+keyword')
+        keyword_column_name = 'hdu+keyword'
+    else:
+        keyword_column_name = 'keyword'
+    # Check how many hdus we are processing
+    final_tables = []
+    for tab in tables:
+        final_table = [table.Column([tab['filename'][0]], name='filename')]
+        for row in tab:
+            final_table.append(table.Column([row['value']], name=row[keyword_column_name]))
+        final_tables.append(table.Table(final_table))
+    final_table = table.vstack(final_tables)
+    # Sort if requested
+    if args.fitsort is not True:  # then it must be a keyword, therefore sort
+        final_table.sort(args.fitsort)
+    # Reorganise to keyword by columns
+    final_table.pprint(max_lines=-1, max_width=-1, show_dtype=True)
+
+
+class KeywordAppendAction(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        if values:
+            keyword = values.replace('.',' ')
+            if namespace.keywords is None:
+                namespace.keywords = []
+            if keyword not in namespace.keywords:
+                namespace.keywords.append(keyword)
+            if option_string in ['-f','--fitsort']:
+                namespace.fitsort = keyword
+        else:
+            if option_string in ['-f','--fitsort']:
+                namespace.fitsort = True
```

Comment:
I don't understand why `fitsort` is managed here. It gives weird result if I use `fitsheader --fitsort astropy/io/fits/tests/data/test1.fits` which seems valid (or should give an error if specifying keywords is required).